src/token.c

```diff
@@ -441,7 +442,10 @@ int allow_insecure_conn(void)
 	//Querying dns for jwt token
 	ret = query_dns(dns_txt_record_id, jwt_token);
 	if(ret){
-		insecure = -1;
+		if (ret == TOKEN_ERR_MEMORY_FAIL)
```

Comment:
Wrap the if statement with {} as lack of these can result in accidental mistakes in the future.